Remove references to any public Staticman API instance

VincentTam opened the pull request:
# Goal

1. Resolves #541
2. Update the instructions for Staticman in the sample config file `_config.yml` so that it complies with Staticman's maintainer's recommendations to
    + use self-hosted instances
    + avoid any public API instances in https://github.com/eduardoboucas/staticman/issues/317#issuecomment-565250060.
3. Seize this chance to announce the end-of-service of my "bot" @<span>staticmanlab</span> on GitHub.

# Description

1. Removes all fallback/default to my Staticman instance `staticman3.herokuapp.com`.
2. Added a brief description about
    1. one-click deploy to Heroku ![Deploy](https://www.herokucdn.com/deploy/button.svg).
    2. GitHub App creation: https://docs.github.com/en/developers/apps/creating-a-github-app, thanks to https://hajekj.net/2020/04/15/staticman-setup-in-app-service/.
    3. Heroku app online dashboard config variables for the Heroku App

        ![screenshot of Heroku's dashboard](https://user-images.githubusercontent.com/5748535/106462435-5e021980-6496-11eb-9c60-05471c067bff.png)

        + `NODE_ENV`: `production`
        + `GITHUB_PRIVATE_KEY`: click the :pen: to paste the content of the RSA private key PEM file _downloaded_  from the GitHub App's settings page.
        + `RSA_PRIVATE_KEY`: click the :pen: to paste the content of your (_locally generated_) RSA private key PEM file.

        ![screenshot of RSA private key on Heroku's dashboard](https://user-images.githubusercontent.com/5748535/106461761-82a9c180-6495-11eb-95ce-72fd8f11f768.png)

    4. Installation of the private GitHub App to the GitHub repo for the Beautiful Jekyll site

# Test

I have only tested my instructions this morning in my Jekyll+Staticman MWE: https://git.io/smdemo: https://github.com/VincentTam/TestStaticmanLab/pull/95.

![Screenshot from 2021-02-01 11-36-53](https://user-images.githubusercontent.com/5748535/106447729-21c4be00-6482-11eb-83d9-c200f4a34a83.png)
![Screenshot from 2021-02-01 11-32-49](https://user-images.githubusercontent.com/5748535/106447733-22f5eb00-6482-11eb-99bc-8232cb96f664.png)

The linked PR in my MWE is "authored" by a "bot", which is the GitHub App "secretman3"

# Critique

I've omitted the instructions for GitLab, for which I opened #510.  Without a project wiki and with Staticman instructions removed from README, there're only two places left for instructions:

1. comments in the site config file `_config.yml`: contains everything about the Staticman setup except reCAPTCHA
2. comments in the Staticman config file `staticman.yml`: contains reCAPTCHA v2 support for Staticman

I've put everything about the GitHub App setup in `_config.yml`.  I'm worried that for users who don't use Staticman, it might be to much to have to another section about the GitLab collaborator bot.  I leave this out for the moment.

daattali commented:
Are all those instructions necessary to have here? If all is needed is the three parameters repository,branch,endpoint then can we provide a link to staticman documentation that says how to get this set up?

VincentTam commented:
> Are all those instructions necessary to have here?

One thing is certain: there's a strong need to change the current instructions, which makes reference to a public accessible API instance associated with @<span>staticmanlab</span>, due to the linked advice from Staticman's maintainer.  I'm really sorry for my lack of action for one whole year.

There're three possibilities:

- GitLab (v3): use bot authorization
- GitHub:
    - v1 & v2 (legacy): use bot authorization
    - v3 : use GitHub App authorization

Our current instructions cover the 2nd one.  I intend to change to the 3rd option, as the 2nd one is legacy, and I think users here are more concerned about deployment on GitHub than on GitLab.

:information_source: I'm happy to answer any questions about the functional aspects of Staticman on either (self-hosted and/or official) GitHub and/or GitLab.  (For technical aspects, my lack of knowledge in node.js might hinder me.)

> If all is needed is the three parameters repository,branch,endpoint then can we provide a link to staticman documentation that says how to get this set up?

It's coming sooner or later: https://github.com/eduardoboucas/staticman.net/pull/13.

daattali commented:
Good to see documentation is going to get updated on the official site! Once the staticman site has documentation for v3, can we just expose the required parameters in _config.yml and link to staticman's website for full explanation of how to get it running?

VincentTam commented:
At the first read, your suggestion makes sense.  Looking back, my first edit of `_config.yml` is too bulky.  I'm going to reduce the instructions in `_config.yml` to two lines.  The official repo, which is maintained by @<span>alexwaibel</span>, has more updated instructions, so we can just refer users to their repo.

```yml
# To use Staticman comments, it's recommended to set up your own API instance.
# Please consult https://github.com/eduardoboucas/staticman/ for server-side setup.
```

I have to explain the client-side parameters in `_config.yml` because they're theme-specific.  I also added some comments in `statciman.yml` concerning the usage of reCAPTCHA v2 with Staticman so as to minimize the ambiguity.  Besides, as the code (in PR #<span>440</span>) is ported from @<span>mmistakes</span>'s Staticman integration for his Jekyll theme Minimal Mistakes, something might work in his way, say the inclusion of the rightmost slash `/`.

```yml
#  endpoint   : # (optional) URL of your own deployment, with a trailing slash eg. https://<your-api>/v3/entry/github/
```

I prefer leaving the explanation with an example to make sure that it's clear.  That can save users from getting errors.  The prime concern of this PR is to remove occurrences of `staticman3 dot herokuapp dot com` (as a fallback/default).  The reason is two-folded:

1. Setting up a personal API instance becomes an official recommendation.
2. I've removed the GitHub and GitLab tokens in `staticman3 dot herokuapp dot com` to stop users from using my API instance.  That stops @<span>staticmanlab</span> from functioning.  Consequently, I have the responsibility to announce this breaking change and offer a fix (a little illustrated guide for setting up Staticman v3 with private GitHub App) for that.

daattali commented:
Where is the official recommendation to use API instances, is it existing yet or is that coming in the PR you linked to above? On the staticman official website I still see instructions saying to add staticman's bot as a collaborator https://staticman.net/docs/getting-started.html

daattali commented:
I've updated the documentation in config.yml and merged

VincentTam commented:
> Where is the official recommendation to use API instances

The link to Staticman's maintainter's (@<span>alexweibel</span>) comment (in Staticman issue #<span>317</span>) on top of the description of this PR.

daattali commented:
Ah I see! You really keep on top of staticman :) I would consider that official when it's properly documented somewhere, as long as it's only mentioned as a comment in an issue/PR I wouldn't consider that an official recommendation

VincentTam commented:
Thanks for clarifying the use of the word "official".  I'm still learning to get into the domain.

Update: the PR (submitted by Staticman's maintainer) that updates the docs on Staticman's website eduardoboucas/staticman.net<span>#</span>13 got approved by Staticman's owner a few hours ago.  We're going to have official doc about Staticman v3 sooner or later.

VincentTam commented:
Update again: the aforementioned PR finally gets merged.  Staticman v3 is now officially documented.
